src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -445,16 +447,21 @@ def generate_config(
         FSID = self.mgr._cluster_fsid
 
         clusters_credentials = {}
-        multi_cluster_config_raw = str(self.mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
-        multi_cluster_config_str = multi_cluster_config_raw.replace("'", '"')
-        valid_multi_cluster_config_str = multi_cluster_config_str.replace('True', '"True"').replace('False', '"False"')
-        multi_cluster_config = json.loads(valid_multi_cluster_config_str)
-        if multi_cluster_config:
-            for url in targets:
-                credentials = self.find_prometheus_credentials(multi_cluster_config, url)
-                if credentials:
-                    clusters_credentials[url] = credentials
-                    clusters_credentials[url]['cert_file_name'] = ''
+        multi_cluster_config_str = str(self.mgr.get_module_option_ex('dashboard', 'MULTICLUSTER_CONFIG'))
+        try:
+            multi_cluster_config = literal_eval(multi_cluster_config_str)  # convert python string to dict
+        except ValueError:
+            multi_cluster_config = None
+            logger.error("Invalid JSON format for multi-cluster config")
```

Comment:
if we're going to raise an error saying it's invalid JSON format, why aren't we using json to load it in?

Reply:
As I mentioned above, we need to use ast.literal_eval here to parse the json. Also since the multi-cluster config can be updated manually as well, hence this check is here

Reply:
I get that, but my point is more if we need to use this special `literal_eval` instead of `json.loads` it seems to imply that the object is not actually proper JSON format.